### ISSUE TITLE
Solver return status

### ISSUE THREAD

**Issue by d7919** [boutproject/BOUT-dev#329]
Currently most solvers typically ignore the return value of `run_rhs` and likewise the `solver` parent class does not use the status code provided by `run` inside the main `solve` call. 

It would be useful if we could update the `solver` parent and the different impls to ensure that we can pass this status through the various calls, this would in principle allow models to indicate they want to stop (e.g. if `max(field)>1e20`) without having to throw an exception. 

There's currently a work around where the user can attach a timestep monitor which does the required checking/decision making as the return status of the monitor **is** usually acted on.

**Comment by dschwoerer** [comment]
I would be in favor of changing the signature and remove the return value.
Returning a non zero value to indicate an error isn't really C++ style.

I would rather add a function to set the flag, provided by #343 to request an exit.
This has not only the advantage that we don't need to change all solver, but also right know I assume in the physics model you set a flag, to change the return value. Directly changing the flag would be in my opinion favorable.

**Comment by bendudson** [comment]
Agree, probably best to remove return code and just throw exception. A breaking change to the API, so for 5.0
